app/models/post.rb

```diff
@@ -0,0 +1,3 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'user_id'
```

Comment:
- [ ] In this `Post` class you should create `update_post_counter` method for the 
- [ ] `Post model` should have a method that returns the 5 most recent comments for a given post

Reply:
### Optional
- [ ] I highly suggest that you use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to database, then make the `update_post_counter` method private

Reply:
> * [ ]  If a post belongs_to an author then foreign_key should be `author_id` not `user_id` please address that

Thanks for your review. I however, do not think there is anything wrong with this. The foreign key can be called anything as long  I make the right references at the application level. And it's only fitting I call it user_id because the value in this column comes from the Users table.

Reply:
@rtonata88 you are right if you make the right reference you can use `user_id` I just suggested it because most of the time it was causing error to other students when creating a database.

Reply:
At this point the database is already created. and I have inserted sample data with no issue. Changing that means I have to go back and redo work of a project approved last week.